### PR TITLE
feat: BlockchainDbMeta builder

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Ref #21 

## Solution

Adds builder methods to instantiate `BlockchainDbMeta`

```rust
 let block =
        provider.get_block(BlockId::latest(), BlockTransactionsKind::Hashes).await?.unwrap();

 let meta = BlockchainDbMeta::default()
        .with_chain_id(31337)
        .with_block(block.inner)
        .with_url(anvil.endpoint());
 ```

## PR Checklist

- [ ] Added Tests
- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Comment by yash-atreya** [comment]
This is not breaking, we keep the `BlockchainDbMeta::new()` method as is.

File: src/cache.rs [hunk 1]
```diff
@@ -133,6 +137,52 @@ impl BlockchainDbMeta {
...
+    /// Sets the chain_id in the [CfgEnv] of this instance.
+    ///
+    /// Remaining fields of [CfgEnv] are left unchanged.
+    pub const fn with_chain_id(mut self, chain_id: u64) -> Self {
+        self.cfg_env.chain_id = chain_id;
+        self
+    }
+
+    /// Sets the [BlockEnv] of this instance using the provided [alloy_rpc_types::Block]
+    pub fn with_block<T: TransactionResponse>(mut self, block: alloy_rpc_types::Block<T>) -> Self {
```
Comment (mattsse) [inline src/cache.rs:150]:
```suggestion
    pub fn with_block<T: TransactionResponse>(mut self, block: &alloy_rpc_types::Block<T>) -> Self {
```

File: src/cache.rs [hunk 2]
```diff
@@ -133,6 +137,52 @@ impl BlockchainDbMeta {
...
+            blob_excess_gas_and_price: Some(BlobExcessGasAndPrice::new(
+                block.header.excess_blob_gas.map(|gas| gas as u64).unwrap_or_default(),
+            )),
+        };
+
+        self
+    }
+
+    /// Infers the host from the provided url and adds it to the set of hosts
+    pub fn with_url(mut self, url: String) -> Self {
```
Comment (mattsse) [inline src/cache.rs:168]: this should accept &str
```suggestion
    pub fn with_url(mut self, url: &str) -> Self {
```

**Review by mattsse — APPROVED** [review]
type nits

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
idk what's up with the new map changes or how to fix @DaniPopes

**Review by mattsse — APPROVED** [review]

File: src/backend.rs [hunk 3]
```diff
@@ -68,8 +68,9 @@ type FullBlockSender =
     OneshotSender<DatabaseResult<WithOtherFields<Block<WithOtherFields<Transaction>>>>>;
 type TransactionSender = OneshotSender<DatabaseResult<WithOtherFields<Transaction>>>;
 
-type AddressData = Map<Address, AccountInfo>;
-type StorageData = Map<Address, StorageInfo>;
+use alloy_primitives::map::AddressHashMap;
```
Comment (DaniPopes) [inline src/backend.rs:71]: please format the import with the other imports

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Pull request merged**